tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -51,6 +65,14 @@ TRTDeviceAllocator::TRTDeviceAllocator(tensorflow::Allocator* allocator)
 
 void TRTDeviceAllocator::free(void* memory) {
   VLOG(2) << "Deallocating @ " << memory;
+  // allocated memory adjusted for alignment, restore the original pointer
+  if (mem_map.count(memory) != 0) {
+    auto alloc_mem = mem_map[memory];
+    mem_map.erase(memory);
+    memory = alloc_mem;
+  }
+  CHECK(mem_pool.count(memory) != 0);
```

Comment:
Is `mem_pool` added just for validation purpose? If yes, how about using `mem_map` to track mappings from aligned address to allocated address regardless whether they're the same (currently `mem_map` only tracks address pairs whose key and value are different)?